### ISSUE TITLE
'Missing client_id or access_token URL parameter.'

### ISSUE THREAD

**Issue by smithyco** [vinkla/instagram#23]
Trying to set up a new Instagram connection in Lumen 5.2. Set up config file, tried a test route, got this:

```
OAuthParameterException in AbstractAdapter.php line 69:
Missing client_id or access_token URL parameter.
```

**Comment by vinkla** [comment]
What version are you using?

**Comment by smithyco** [comment]
@vinkla 
Version of Lumen: 5.2
Version of Vinkla/Instagram: 4.0

**Comment by vinkla** [comment]
@hassankhan do you maybe know what this could be?

**Comment by vinkla** [comment]
@smithyco have you added your Instagram config correctly?

**Comment by smithyco** [comment]
Yep, I added the client ID and the client secret. I can't find the "access token", I didn't think that was needed. I also just tested the same version of the code on a fresh Laravel install, and I'm still getting the 'Missing client_id or access_token URL parameter.'

**Comment by hassankhan** [comment]
Seems like no access token has been set, you'll need that to make requests.

**Comment by smithyco** [comment]
It looks like the wrapper provided here doesn't retrieve that access token and set it in the client if a null value is provided.

**Comment by hassankhan** [comment]
That's odd, after calling `getAccessToken()` it should set the access token. I'll have a look and see if I can figure out what's going on.

**Comment by valmont137** [comment]
I have same error on fresh Laravel 5.2 installation. Maybe it's oAuth problem, because manualy calling `getAccessToken()` return `Exception in NativeSessionStore.php line 28: Sessions are not active.`

**Comment by vinkla** [comment]
@hassankhan maybe we need to add a custom session storage?

**Comment by hassankhan** [comment]
Ahh yeah, did you not add one?

**Comment by vinkla** [comment]
@hassankhan working on it #24

**Comment by vinkla** [comment]
@smithyco @valmont137 I've pushed an update in 4.0.1 Could you guys please test?

**Comment by valmont137** [comment]
`Instagram::users()->get(30);` still return `'Missing client_id or access_token URL parameter.'`

manually set access_token in config return 

```
ErrorException in AccessToken.php line 55
at HandleExceptions->handleError('4096', 'Argument 1 passed to League\OAuth2\Client\Token\AccessToken::__construct() must be of the type array, null given, called in \vendor\larabros\elogram\src\Providers\MiddlewareServiceProvider.php on line 56 and defined', '\vendor\league\oauth2-client\src\Token\AccessToken.php', '55', array()) in AccessToken.php line 55
```
or by `getAccessToken()` 

```
in SessionStore.php line 36
at HandleExceptions->handleError('4096', 'Argument 1 passed to Vinkla\Instagram\Session\SessionStore::__construct() must be an instance of Illuminate\Session\Store, none given, called in \vendor\larabros\elogram\src\Providers\CoreServiceProvider.php on line 60 and defined', '\vendor\vinkla\instagram\src\Session\SessionStore.php', '36', array()) in SessionStore.php line 36
```

maybe I do something wrong?

**Comment by vinkla** [comment]
@valmont137 no, this is our fault. We'll try to find a fix.

**Comment by valmont137** [comment]
`Instagram::getLoginUrl(); ` alse return `ErrorException in SessionStore.php line 36`

I tried use example from http://elogram.readthedocs.org/en/stable/quickstart.html#creating-a-client without Instagram wrapper - it's also `Sessions are not active.`

**Comment by hassankhan** [comment]
My bad, the docs should also state that you should manually call `session_start()` in your own code, if not using a framework.

**Comment by kamisaa** [comment]
im using elogram too and get same problem 

`Fatal error: Uncaught exception 'Larabros\Elogram\Exceptions\Exception' with message 'Sessions are not active.' in /var/www/html/composer/vendor/larabros/elogram/src/Http/Sessions/NativeSessionStore.php:28`

**Comment by smithyco** [comment]
```
Route::get('/instagram', function()
{
    session_start();
    $client = new \Larabros\Elogram\Client(
        env('INSTAGRAM_ID'),
        env('INSTAGRAM_SECRET'),
        null,
        env('INSTAGRAM_REDIRECT')
    );
    // If we don't have an authorization code then get one
    if (!isset($_GET['code'])) {
        header('Location: ' . $client->getLoginUrl());
        exit;
    } else {
        $token = $client->getAccessToken($_GET['code']);
    }
    return $client->users()->search('skrawg');
});
```
This returns:
```
FatalThrowableError in AuthMiddleware.php line 36:
Fatal error: Call to a member function getToken() on string
```
It looks like the token is being set as a string in the config array, and then it cannot be converted back into the AccessToken class to make the method call.

**Comment by hassankhan** [comment]
Just pushed a fix to Elogram - mind running `composer update` again?

**Comment by smithyco** [comment]
That worked! It's pulling data in the above code block now.

With a laravel route, I thought that I wouldn't have to kick off a `session_start()`. Is that not the case? Or do I need to wrap it in some middleware?

**Comment by hassankhan** [comment]
I've made a PR to *this* package which properly integrates it with Laravel, just waiting on @vinkla to review and merge, then you should be able to remove the `session_start()` and use this package as intended :smile:

**Comment by kdeit** [comment]
Laravel 5.2. Have some error
`ErrorException in AccessToken.php line 55:
Argument 1 passed to League\OAuth2\Client\Token\AccessToken::__construct() must be of the type array, null given, called in `

**Comment by alberto-bottarini** [comment]
Don't ask me about this error, but you need to change your instagram.php in this way:

```
'main' => [
            'id' => env('INSTAGRAM_CLIENT_ID'),
            'secret' => env('INSTAGRAM_CLIENT_SECRET'),
            'access_token' => '{ "access_token": "' . env('INSTAGRAM_ACCESS_TOKEN') . '" }',
        ]
```

Access Token property must be a json string that will be parsed by php files.

**Comment by hassankhan** [comment]
Hi guys, this problem should be resolved once #25 is merged

**Comment by Baadier-Sydow** [comment]
Any update on this or a fix that works in the interim?

**Comment by kdeit** [comment]
The solution is 
`'main' => [
            'id' => 'some_secret_id',
            'secret' => 'some_secret',
            'access_token' => '{ "access_token": "some_secret_token"}'
        ],`

**Comment by Baadier-Sydow** [comment]
@kdeit Is it not supposed to generate the access token from the client id and secret which it doesnt appear to do?

**Comment by berkayk** [comment]
I am having the same problem. If I get the code and call Instagram::getAccessToken($code), it throws **Sessions are not active** error.

**Comment by valentincognito** [comment]
Tried all the above still doesn't work... Now I get 
APINotFoundError in AbstractAdapter.php line 69:
this user does not exist

Anyone else has this error ?

I am running laravel5.2 and vinkla/instagram 4.0

**Comment by hassankhan** [comment]
@valentincognito Can you open an issue [here](https://github.com/larabros/elogram) please with some more details? Thanks :)

**Comment by callmedela** [comment]
I am not able to use your package, this is what i get
![screen shot 2016-05-31 at 4 47 38 am](https://cloud.githubusercontent.com/assets/8988998/15663462/e6feb778-26ea-11e6-9a1a-760e3b838fba.png)

**Comment by vinkla** [comment]
@callmedela please open a new issue.

**Comment by mohsenweb** [comment]
@vinkla i have a problem like @callmedela

**Comment by vinkla** [comment]
Okay, sounds far. We're going to implement a fix soon. Right now, there is to much to do.

**Comment by hassankhan** [comment]
Hi all, this issue is dependent on https://github.com/larabros/elogram/issues/40, please be patient. Thanks 😄

**Comment by alexschwarz89** [comment]
It is actually possible to do a request with some configuration? If i specify `null` for access_token i get "Missing client_id or access_token URL parameter.". 

When i specify an access_token in a JSON string like mentioned above, i get "The access_token provided is invalid.". Double checked the token received from https://api.instagram.com/oauth/authorize/. The app is still in sandbox mode. Any idea?

**Comment by HarryBosh** [comment]
So this is basically unusable? you need to state that on the readme.md at the very top!

**Comment by vinkla** [comment]
The package is now rewritten and doesn't use the Instagram API no more.
